commands/deploy.go

```diff
@@ -292,8 +293,9 @@ Error: %s`, fprocessErr.Error())
 				fmt.Println(msg)
 			}
 
-			statusCode := proxy.DeployFunction(deploySpec)
-
+			cliAuth := NewCLIAuth(token, gateway)
+			proxyClient := proxy.NewClient(cliAuth, gateway, deploySpec.TLSInsecure)
+			statusCode := proxyClient.DeployFunction(context.TODO(), deploySpec)
```

Comment:
Should we use TODO() vs Background()?

Reply:
TODO indicates "placeholder for future _accurate_ context" while background indicates that this is a main thread or somehow that the parent context should not be connected.  TODO seems correct to me

https://golang.org/pkg/context/#Background

> Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

Reply:
We use context.Background() everywhere else and I see it used a lot in K8s/Docker. What's the main downside, should we be changing this everywhere else it is used?

Reply:
I made a typo in my previous comment, I didn't mean TODO made sense here, but instead we should use Background.  Since this is the implementation of the cmd run, this is essentially the main function

Reply:
@viveksyngh i think we want to switch this to Background here
```suggestion
			statusCode := proxyClient.DeployFunction(context.Background(), deploySpec)
```

Reply:
yes .. I will do that.